MinimedKit/GlucosePage.swift

```diff
@@ -69,8 +68,10 @@ public class GlucosePage {
                 let event = matchEvent(tempOffset, relativeTimestamp: DateComponents())
                 if event is RelativeTimestampedGlucoseEvent {
                     relativeEventCount += 1
-                } else if let event = event as? ReferenceTimestampedGlucoseEvent {
-                    let offsetDate = calendar.date(byAdding: Calendar.Component.minute, value: 5 * relativeEventCount, to: event.timestamp.date!)!
+                } else if let sensorTimestampEvent = event as? SensorTimestampGlucoseEvent,
+                    relativeEventCount == 0 || (event as! SensorTimestampGlucoseEvent).isForwardOffsetReference() {
+                    
+                    let offsetDate = calendar.date(byAdding: Calendar.Component.minute, value: 5 * relativeEventCount, to: sensorTimestampEvent.timestamp.date!)!
```

Comment:
`byAdding: .minute,` since the type is inferred.

Also, I'm not sure this line to be a `calendar` operation if you're just adding a double to a double (the next one absolutely does)
`sensorTimestampEvent.timestamp.date?.addingTimeInterval(TimeInterval(minutes: 5 * relativeEventCount))`